sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/ServerSentEvent.java

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.models;
+
+import java.time.Duration;
+import java.util.List;
+
+/**
+ * Represents the SSE response from the server on event stream interpretation
+ * <a href="https://html.spec.whatwg.org/multipage/server-sent-events.html#parsing-an-event-stream">here</a>.
+ */
+public final class ServerSentEvent {
+    private Long id;
+    private String event;
+    private List<String> data;
+    private String comment;
+    private Duration retryAfter;
```

Comment:
I am not sure the end user will be able to use `retryAfter` as this should be handled by the core library (or the client library). Since this is a user-facing type, we should remove it. If it's needed and we have a use case for it, we can add it back later.